Consider initContainer GPUs quota in calculating

kerthcet opened the pull request:
Signed-off-by: kerthcet <kerthcet@gmail.com>

fix: https://github.com/volcano-sh/volcano/issues/2422

Thor-wl commented on pkg/scheduler/api/device_info.go:
```diff
@@ -55,15 +55,27 @@ func (g *GPUDevice) getUsedGPUMemory() uint {
 
 // getGPUResourceOfPod returns the GPU resource required by the container.
-func getGPUResourceOfContainer(container *v1.Container) uint {
+func getGPUResourceOfContainer(container v1.Container) uint {
```
Suggest to keep the object reference as the input parameter for we only need the `spec.resource` part. That will be more efficient.

kerthcet replied:
Make sense, currently we pass a value here.

kerthcet commented:
Updated and squashed.
The flaky test maybe not related to this PR.
```
failed to create job queue2-job in namespace 
  Unexpected error:
      <*errors.StatusError | 0xc00044e960>: {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "admission webhook \"validatejob.volcano.sh\" denied the request: can only submit job to queue with state `Open`, queue `queue2` status is ``;",
              Reason: "",
              Details: nil,
              Code: 400,
          },
      }
      admission webhook "validatejob.volcano.sh" denied the request: can only submit job to queue with state `Open`, queue `queue2` status is ``;
  occurred
```

hwdef reviewed (commented):
/lgtm
Thanks!

Thor-wl commented:
@kerthcet Hi, Could you cherry pick this patch to branch v1.6/v.15/v1.4? It makes sense.

kerthcet commented:
I think you mean release-1.6, release-1.5, release-1.4, right?

Thor-wl commented:
> I think you mean release-1.6, release-1.5, release-1.4, right?

Yes.
